logos_setup.py

```diff
@@ -0,0 +1,801 @@
+    if VERBOSE:
+        print(datetime.datetime.now())
+    chooseVersion()  # We ask user for his Faithlife product's version, set variables, and create project skeleton.
+    if VERBOSE:
+        print(datetime.datetime.now())
+    getLogosReleaseVersion(getLogosReleases())
+    if VERBOSE:
+        print(datetime.datetime.now())
+    logos_setup() # We set some basic variables for the install, including retrieving the product's latest release.
+    if VERBOSE:
+        print(datetime.datetime.now())
+    chooseInstallMethod()  # We ask user for his desired install method.
+    # END PREPARATION
+    
+    TARGETVERSION = os.environ.get('TARGETVERSION')
+    LOGOS_EXE = os.environ.get('LOGOS_EXE')
+    FLPRODUCT = os.environ.get('FLPRODUCT')
+    WINEPREFIX = os.environ.get('WINEPREFIX')
+
+    if REGENERATE == "":
+        if VERBOSE:
+            print(datetime.datetime.now())
+        if checkExistingInstall():
+            logos_error("Existing installation.")
+        
+        if VERBOSE:
+            print(datetime.datetime.now())
+        beginInstall()
+
+        if VERBOSE:
+            print(datetime.datetime.now())
+        initializeWineBottle()  # We run wineboot.
+        
+        if TARGETVERSION == "10":
+            if VERBOSE:
+                print(datetime.datetime.now())
+            installLogos10()  # We run the commands specific to Logos 10.
+        elif TARGETVERSION == "9":
+            if VERBOSE:
+                print(datetime.datetime.now())
+            installLogos9()  # We run the commands specific to Logos 9.
+        else:
+            logos_error(f"TARGETVERSION unrecognized: '{TARGETVERSION}'. Installation canceled!", "")
+        
+        if VERBOSE:
+            print(datetime.datetime.now())
+        create_starting_scripts()
+        
+        heavy_wineserver_wait()
+        
+        clean_all()
+        
+        exes = [e for e in glob.glob(f"{WINEPREFIX}/drive_c/**/{FLPRODUCT}.exe", recursive=True) if 'Pending' not in e]
+        LOGOS_EXE = exes[0]
+        
+        if VERBOSE:
+            print(datetime.datetime.now())
+        postInstall()
+    else:
+        create_starting_scripts()
+        logos_info("The scripts have been regenerated.")
+    
+    # with open(LOGOS_LOG, "a") as f:
+    #     f.write(output) #FIXME: What should be in "output"?
+
+def set_appimage(newappimage):
+    WINE64_APPIMAGE_FULL_VERSION = os.environ["WINE64_APPIMAGE_FULL_VERSION"]
+    WINE64_APPIMAGE_FULL_URL = os.environ["WINE64_APPIMAGE_FULL_URL"]
+    WINE64_APPIMAGE_FULL_FILENAME = os.environ["WINE64_APPIMAGE_FULL_FILENAME"]
+    WINE64_APPIMAGE_VERSION = os.environ["WINE64_APPIMAGE_VERSION"]
+    WINE64_APPIMAGE_URL = os.environ["WINE64_APPIMAGE_URL"]
+    WINE64_APPIMAGE_FILENAME = os.environ["WINE64_APPIMAGE_FILENAME"]
+    APPIMAGE_LINK_SELECTION_NAME = os.environ["APPIMAGE_LINK_SELECTION_NAME"]
+    subprocess.run(["ln", "-s", SET_APPIMAGE_FILENAME, f"{APPDIR_BINDIR}/{APPIMAGE_LINK_SELECTION_NAME}"])
```

Comment:
This can go to utils.py

Reply:
Done in [6f3b432](https://github.com/thw26/LogosLinuxInstaller/pull/7/commits/6f3b4325ec22de4d31eece488ae2a41a7f029e45)